activerecord/test/cases/serialized_attribute_test.rb

```diff
@@ -41,6 +41,16 @@ def test_serialized_attribute
     assert_equal(myobj, topic.content)
   end
 
+  def test_serialized_attribute_with_default
+    klass = Class.new(ActiveRecord::Base) do
+      self.table_name = Topic.table_name
+      serialize(:content, Hash, default: { key: "value" })
+    end
+
+    t = klass.new
+    assert_equal({ key: "value" }, t.content)
+  end
```

Comment:
Can you also add a test for `serialize` on custom attribute with default?

```ruby
  def test_serialized_attribute_on_custom_attribute_with_default
    klass = Class.new(ActiveRecord::Base) do
      self.table_name = Topic.table_name
      attribute :content, default: { key: "value" }
      serialize :content, Hash
    end

    t = klass.new
    assert_equal({ key: "value" }, t.content)
  end
```

We've discussed about fixing the regression 8f2decdf4db9723dd964637cb6357f0aabcb4e24, and have agreed that allowing default to be configured with `serialize` is also useful, similar to #39820.

Reply:
I just added 👍